crates/driver/src/domain/competition/mod.rs

```diff
@@ -52,11 +54,44 @@ pub struct Competition {
     pub mempools: Mempools,
     /// Cached solutions with the most recent solutions at the front.
     pub settlements: Mutex<VecDeque<Settlement>>,
+    settle_queue: mpsc::Sender<SettleRequest>,
 }
 
 impl Competition {
+    pub fn new(
+        solver: Solver,
+        eth: Ethereum,
+        liquidity: infra::liquidity::Fetcher,
+        simulator: Simulator,
+        mempools: Mempools,
+    ) -> Arc<Self> {
+        let (settle_tx, settle_rx) = mpsc::channel(solver.settle_queue_size());
+
+        let competition = Arc::new(Self {
+            solver,
+            eth,
+            liquidity,
+            simulator,
+            mempools,
+            settlements: Default::default(),
+            settle_queue: settle_tx,
+        });
+
+        let competition_clone = Arc::clone(&competition);
+        tokio::spawn(async move {
+            competition_clone.process_settle_requests(settle_rx).await;
+        });
+
+        competition
+    }
+
     /// Solve an auction as part of this competition.
     pub async fn solve(&self, auction: &Auction) -> Result<Option<Solved>, Error> {
+        if self.settle_queue.capacity() == 0 {
+            tracing::warn!("settlement queue is full; auction is rejected");
+            return Err(Error::SettlementQueueIsFull);
+        }
```

Comment:
Would it be more clever to put this `if` right before solve response is sent to autopilot? Because settle queue could be freed by the time this long lasting function is finished?

Reply:
Thanks, yes, that makes sense.